### PR TITLE
kernel/fault_manager: Add support of fault manager for armv7-m

### PR DESCRIPTION
**Pull request by drashti304**
This patch adds support for fault manger for armv7-m architecture.

NOTE:
- This patch adds skeleton code for get_appid() and kill_app()
- In this patch, multi instances of the same application is not supported
with fault manager
- For now assert cases is only supported with fault manager

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [7dee563ea7ccb11d8240b51d7b06f00a409669a8] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [7dee563ea7ccb11d8240b51d7b06f00a409669a8] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [7d824e55706a9de6a4a5454999ffd603d1160a27] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [7d824e55706a9de6a4a5454999ffd603d1160a27] - Code Rule Check OK.

File: os/kernel/fault_manager/fault_manager.h [hunk 1]
```diff
@@ -57,12 +58,13 @@ typedef struct fm_binary_table_s fm_binary_table_t;
...
  *   pid    - The pid of the binary to be registered
  *   result - Binary path to be registered with the fault manager
+ *   bin_size - Binary size to be registered with the fault manager
  *
  * Returned Value:
  *   Zero (OK) on success; otherwise a -1 (ERROR) value is returned.
  *
  ****************************************************************************/
-int fm_register_bianry(pid_t pid, const char *path);
+int fm_register_bianry(pid_t pid, const char *path, int bin_size);
```
Comment (sijupunnnoose) [inline os/kernel/fault_manager/fault_manager.h]: bianry->binary

**Comment by seinfra** [comment]
Target : [8a3969b1737fdd6b4c4286f4a750b499ccf545d7] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [8a3969b1737fdd6b4c4286f4a750b499ccf545d7] - Code Rule Check OK.

File: os/arch/arm/src/armv7-m/up_hardfault.c [hunk 2]
```diff
@@ -148,6 +155,7 @@ int up_hardfault(int irq, FAR void *context, FAR void *arg)
 
 		if (insn == INSN_SVC0) {
 			hfdbg("Forward SVCall\n");
+			lldbg("Forward SVCall\n");
```
Comment (sunghan-chang) [inline os/arch/arm/src/armv7-m/up_hardfault.c]: What is difference between hfdbg above and this lldbg?

File: os/arch/arm/src/armv7-m/up_assert.c [hunk 3]
```diff
@@ -397,8 +407,46 @@ static void up_dumpstate(void)
...
+		}
+#endif
+	} else {
+		fault_queue = mq_open(FAULT_MGR_MQ, O_WRONLY, 0666, 0);
+		DEBUGASSERT(fault_queue != (mqd_t)ERROR);
+
+		pid = getpid();
 
+		/* Send a message to fault manager with pid of the faulty task */
+		ret = mq_send(fault_queue, (const char *)&pid, sizeof(pid), 0);
```
Comment (jeongarmy) [inline os/arch/arm/src/armv7-m/up_assert.c]: please add mq_close(fault_queue); and remove DEBUGASSERT

**Comment by sunghan-chang** [comment]
@drashti304 Could you update this as soon as possible?

**Comment by seinfra** [comment]
Target : [3edffdbb24a5e7c6c82b9f2c1701f385e4b82d21] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3edffdbb24a5e7c6c82b9f2c1701f385e4b82d21] - Code Rule Check OK.

File: os/arch/arm/src/armv7-m/up_hardfault.c (hunk 2 above)
Comment (drashti304) [reply os/arch/arm/src/armv7-m/up_hardfault.c]: Added for testing, I have removed now.
Please check

File: os/arch/arm/src/armv7-m/up_assert.c (hunk 3 above)
Comment (drashti304) [reply os/arch/arm/src/armv7-m/up_assert.c]: Added, thanks

**Comment by seinfra** [comment]
Target : [0a66e8b5915390bfedd73bd221b6f6e05d3435ab] - Code Rule Check OK.

**Comment by drashti304** [comment]
> @drashti304 Could you update this as soon as possible?

done

**Comment by seinfra** [comment]
Target : [0a66e8b5915390bfedd73bd221b6f6e05d3435ab] - Code Rule Check (C++) OK.

**Pull request merged**